packages/http-server/src/__tests__/body-params-validation.spec.ts

```diff
@@ -0,0 +1,520 @@
+            validation: [{ code: 'required', message: "should have required property 'id'", severity: 'Error' }],
+          });
+        });
+      });
+    });
+
+    describe('operation with optional body', () => {
+      describe('when no body provided', () => {
+        test('returns 200', async () => {
+          const response = await server.fastify.inject({
+            method: 'POST',
+            url: '/json-body-optional',
+          });
+
+          expect(response.statusCode).toBe(200);
+        });
+      });
+    });
+
+    describe('operation with required body', () => {
+      const operation: fastify.HTTPInjectOptions = {
+        method: 'POST',
+        url: '/json-body-required',
+      };
+
+      describe('when no body provided', () => {
+        test('returns 422 & error message', async () => {
+          const response = await server.fastify.inject(operation);
+
+          expect(response.statusCode).toBe(422);
+          expect(JSON.parse(response.payload)).toMatchObject({
+            validation: [{ code: 'required', message: 'Body parameter is required', severity: 'Error' }],
+          });
+        });
+      });
+
+      describe('when body provided', () => {
+        describe('and is valid', () => {
+          // Ref: https://github.com/stoplightio/prism/issues/500
+          test.todo('returns 200');
+        });
+
+        describe('and property type invalid', () => {
+          test('returns 422 & error message', async () => {
+            const response = await server.fastify.inject({
+              ...operation,
+              payload: {
+                id: 'string',
+              },
+            });
+
+            expect(response.statusCode).toBe(422);
+            expect(JSON.parse(response.payload)).toMatchObject({
+              validation: [
+                {
+                  code: 'type',
+                  location: ['body', 'id'],
+                  message: 'should be integer',
+                  severity: 'Error',
+                },
+              ],
+            });
+          });
+        });
+
+        describe('and property not one of enum', () => {
+          test('returns 422 & error message', async () => {
+            const response = await server.fastify.inject({
+              ...operation,
+              payload: {
+                status: 'string',
+              },
+            });
+
+            expect(response.statusCode).toBe(422);
+            expect(JSON.parse(response.payload)).toMatchObject({
+              validation: [
+                {
+                  code: 'enum',
+                  location: ['body', 'status'],
+                  message: 'should be equal to one of the allowed values',
+                  severity: 'Error',
+                },
+              ],
+            });
+          });
+        });
+      });
+    });
+  });
+
+  describe.each([['oas2'], ['oas3']])('%s with form data param', oas => {
```

Comment:
Do we  see special oas 2 and 3 variances now that we’re just using operations directly?

Reply:
Omg I freaking missed it :P fixing asap

Reply:
@philsturgeon done